feat: update role ux in projec access list

IrakliJani opened the pull request:
Closes: https://github.com/lightdash/lightdash/issues/8512
Closes: https://github.com/lightdash/lightdash/issues/8499

### Description:

![CleanShot 2024-01-22 at 14 50 19@2x](https://github.com/lightdash/lightdash/assets/962095/9d50cadb-9bb2-4615-b1db-dca88861c2b0)

![CleanShot 2024-01-22 at 14 50 05@2x](https://github.com/lightdash/lightdash/assets/962095/d6d913b7-05c7-4957-af04-b3e8808e8bf9)



### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

magnew reviewed (approved):
Reviewed and tested. I like how this is done. 

I left a non-blocking suggesting for code clarity. If you disagree feel free to leave as is.

IrakliJani commented on packages/frontend/src/components/ProjectAccess/ProjectAccessRow.tsx:
```diff
@@ -68,88 +66,105 @@ const ProjectAccessRow: FC<Props> = ({
+                            }}
+                        />
+
+                        {highestRole.type !== 'project' && projectRole?.role && (
```
that makes sense. thank you!

IrakliJani commented:
@TuringLovesDeathMetal @magnew addressed all the comments! thanks for the review!

![CleanShot 2024-01-22 at 18 49 48](https://github.com/lightdash/lightdash/assets/962095/f15e77ca-2b20-45db-8e20-3303bd349f2e)

TuringLovesDeathMetal reviewed (approved):
😚 👌  amazing - this looks SO much better! Well done Irakli!

IrakliJani commented:
ignoring e2e, is not related to the PR
